examples/mugen/retrieval/s3d.py

```diff
@@ -111,262 +111,42 @@ def forward(self, x):
+            SepConv3d(mid_planes, out_planes, kernel_size=3, stride=1, padding=1),
         )
-        self.branch2 = nn.Sequential(
-            BasicConv3d(480, 16, kernel_size=1, stride=1),
-            SepConv3d(16, 48, kernel_size=3, stride=1, padding=1),
-        )
-        self.branch3 = nn.Sequential(
-            nn.MaxPool3d(kernel_size=(3, 3, 3), stride=1, padding=1),
-            BasicConv3d(480, 64, kernel_size=1, stride=1),
-        )
-
-    def forward(self, x):
-        x0 = self.branch0(x)
-        x1 = self.branch1(x)
-        x2 = self.branch2(x)
-        x3 = self.branch3(x)
-        out = torch.cat((x0, x1, x2, x3), 1)
-        return out
-
 
-class Mixed4c(nn.Module):
-    def __init__(self):
-        super(Mixed4c, self).__init__()
 
-        self.branch0 = nn.Sequential(
-            BasicConv3d(512, 160, kernel_size=1, stride=1),
-        )
-        self.branch1 = nn.Sequential(
-            BasicConv3d(512, 112, kernel_size=1, stride=1),
-            SepConv3d(112, 224, kernel_size=3, stride=1, padding=1),
-        )
-        self.branch2 = nn.Sequential(
-            BasicConv3d(512, 24, kernel_size=1, stride=1),
-            SepConv3d(24, 64, kernel_size=3, stride=1, padding=1),
-        )
-        self.branch3 = nn.Sequential(
+class Branch3(nn.Sequential):
+    def __init__(self, in_planes, out_planes):
+        super(Branch3, self).__init__(
             nn.MaxPool3d(kernel_size=(3, 3, 3), stride=1, padding=1),
-            BasicConv3d(512, 64, kernel_size=1, stride=1),
+            BasicConv3d(in_planes, out_planes, kernel_size=1, stride=1),
         )
 
-    def forward(self, x):
-        x0 = self.branch0(x)
-        x1 = self.branch1(x)
-        x2 = self.branch2(x)
-        x3 = self.branch3(x)
-        out = torch.cat((x0, x1, x2, x3), 1)
-        return out
 
-
-class Mixed4d(nn.Module):
-    def __init__(self):
-        super(Mixed4d, self).__init__()
-
-        self.branch0 = nn.Sequential(
-            BasicConv3d(512, 128, kernel_size=1, stride=1),
-        )
-        self.branch1 = nn.Sequential(
-            BasicConv3d(512, 128, kernel_size=1, stride=1),
-            SepConv3d(128, 256, kernel_size=3, stride=1, padding=1),
-        )
-        self.branch2 = nn.Sequential(
-            BasicConv3d(512, 24, kernel_size=1, stride=1),
-            SepConv3d(24, 64, kernel_size=3, stride=1, padding=1),
-        )
-        self.branch3 = nn.Sequential(
-            nn.MaxPool3d(kernel_size=(3, 3, 3), stride=1, padding=1),
-            BasicConv3d(512, 64, kernel_size=1, stride=1),
-        )
+class MixedConvsBlock(nn.Module):
+    def __init__(self, in_planes, b0_out, b1_mid, b1_out, b2_mid, b2_out, b3_out):
+        super().__init__()
+        self.branch0 = Branch0(in_planes, b0_out)
+        self.branch1 = Branch1or2(in_planes, b1_mid, b1_out)
+        self.branch2 = Branch1or2(in_planes, b2_mid, b2_out)
+        self.branch3 = Branch3(in_planes, b3_out)
```

Comment:
Can we avoid the introduction of the `Branch*` classes? From what I understand, we can just use Sequential directly and simplify the code structure.